### ISSUE TITLE
OS_AUTH_URL

### ISSUE THREAD

**Issue by jmamedov** [OpenStackCookbook/OpenStackCookbook#57]
Following instructions on the the book: Unable to create tenant.
First, your instruction "keystone tenant-create ....." displays error
"The request you have made requires authentication. (HTTP 401)"
Then tried with "sudo keystone tenant-create..." and another error:
"Expecting an auth URL via either --os-auth-url or env[OS_AUTH_URL]"
I have set up all the variables (following book instruction).
By the way I am using your Vargant files.

**Comment by uksysadmin** [comment]
Have you sourced in your credentials?

source /vagrant/openrc

Then you can do the administration and operation of OpenStack.
(Also don't use sudo, that would launch a new shell, where your credentials (env variables) won't exist).

**Comment by jmamedov** [comment]
I just did. That statement is not in the book as an instruction.
Got the same error: HTTP (401)

**Comment by jmamedov** [comment]
Here is a screen shot:
vagrant@controller:~$ export OS_TENANT_NAME=cookbook
vagrant@controller:~$ export OS_USERNAME=admin
vagrant@controller:~$ export OS_PASSWORD=openstack
vagrant@controller:~$ export OS_AUTH_URL=https://192.168.100.200:5000/v2.0/
vagrant@controller:~$ export OS_NO_CACHE=1
vagrant@controller:~$ export OS_KEY=/vagrant/cakey.pem
vagrant@controller:~$ export OS_CACERT=/vagrant/ca.pem
vagrant@controller:~$ source /vagrant/openrc
vagrant@controller:~$ keystone tenant-create --name cookbook --description "Def
ault Cookbook Tenant" --enable true
The request you have made requires authentication. (HTTP 401)

**Comment by uksysadmin** [comment]
401 is access denied. This is saying the service is up, but your credentials aren't allowing you to do this which implies the admin user, with password openstack in the tenant admin hasn't been created. Check the keystone logs on the controller for errors. I'd use the SERVICE_TOKEN from /etc/keystone/keystone.conf and the SERVICE_ENDPOINT to interrogate Keystone to check the users. So set those Env variables, then do a keystone user-list; keystone tenant-list. It's looking likely the vagrant up mustn't have completed properly.

**Comment by jmamedov** [comment]
I don't see SERVICE_TOKEN or SERVICE_ENDPOINT qualifiers in keystone.conf

**Comment by uksysadmin** [comment]
How much Ram do you have? To successfully run this, we use 16Gb machines. 8Gb (on a Mac and VirtualBox seems to struggle). The reason is that the scripts work, but some transient error has caused the Keystone installation to fail, which subsequently has probably caused other issues that you've not encountered yet. As its vagrant - I'd start again: vagrant destroy -f; vagrant up.

**Comment by jmamedov** [comment]
Well, I have a lot of memory 32 GB in my laptop. I think I need to star over as you mentioned.

**Comment by uksysadmin** [comment]
Yes. Before you do: vagrant status and check if all machines were up. The controller (and Keystone) gets done first. There will be errors in the output that can easily get missed. I tend to do: 'vagrant up | tee -a vagrant.out' so I can grep through the output on success or failure.

**Comment by jmamedov** [comment]
Ok. I guessing after destroy I need to git it again.

**Comment by jmamedov** [comment]
Never mind. It did stat and hopefully it will work at this time.
Thanks.

**Comment by jmamedov** [comment]
==> network: Package configuration┌──────────────────────────┤ Configuring grub-
pc ├──────────────────────────┐│││ The GRUB boot loader was previously installed
 to a disk that is no││ longer present, or whose unique identifier has changed f
or some reason.   ││ It is important to make sure that the installed GRUB core i
mage stays in  ││ sync with GRUB modules and grub.cfg. Please check again to mak
e sure││ that GRUB is written to the appropriate boot devices.││││ If you're uns
ure which drive is designated as boot drive by your BIOS,    ││ it is often a go
od idea to install GRUB to all of them.││││ Note: it is possible to install GRUB
 to partition boot records as well,   ││ and some appropriate partitions are off
ered here. However, this forces    ││ GRUB to use the blocklist mechanism, which
 makes it less reliable, and    ││ therefore is not recommended.││││<Ok>│││└────
───────────────────────────────────────────────────────────────────────┘

==> network: Failed to open terminal.
==> network: debconf: whiptail output the above errors, giving up!
==> network: dpkg: error processing package grub-pc (--configure):
==> network:  subprocess installed post-installation script returned error exit
status 255
==> network: Errors were encountered while processing:
==> network:  grub-pc
==> network: E
==> network: :
==> network: Sub-process /usr/bin/dpkg returned an error code (1)

**Comment by uksysadmin** [comment]
Let me know the versions of Vagrant and VirtualBox, and on what OS?

**Comment by jmamedov** [comment]
Vagrant version = 1.7.4
VB version = 5.0.4

**Comment by jmamedov** [comment]
OS = Win 10 64 bit

**Comment by jmamedov** [comment]
I have output log but, this will not accept it as an attachment.
Will email to you.

**Comment by jmamedov** [comment]
Output of 'vagrant up' command

**Comment by uksysadmin** [comment]
We haven't done that much testing of Windows as the base host. All development is done using OSX or Ubuntu. You may be uncovering issues we didn't cater for/develop for. I'll check tomorrow on a Windows PC.

**Comment by jmamedov** [comment]
I am using Windows as a host box that Virtual Box installed and there are guest VM(s) (Linux, Solaris, others) are running fine.
Want to make I am clear what you are saying. Which is your development has not been tested on host server as a Windows operating system.

**Comment by uksysadmin** [comment]
Vagrant has some OS specific sections that are only required for Windows as OSX and Linux tend to be more alike for the vast majority of basic Vagrant scripts. What I'm saying is that YMMV with the environment you're using, but we welcome forks and pull requests.
